<launch>
    <arg name="output" value="screen"/>
    <arg name="hist_name" default="MyHist"/>
    <arg name="hist_path" default="$(find extended_object_detection)/config/object_base_example/histograms/$(arg hist_name).yaml"/>
    
    <node name="camera" pkg="usb_cam" type="usb_cam_node" output="$(arg output)" clear_params="true" respawn="false" required="false" launch-prefix="" >
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="framerate" value="30" />
        <param name="camera_frame_id" value="cam_optical_link" />
        <param name="pixel_format" value="yuyv" />
        
        <param name="camera_info_url" value="file://$(find extended_object_detection)/config/DefenderC-090.yaml" />
        <param name="camera_name" value="camera"/>
    </node>
    
    <node name="hist_color_params_collector_contour_node" pkg="extended_object_detection" type="hist_color_params_collector_contour_node" output="screen" required="true">        
        <param name="out_filename" value="$(arg hist_path)"/>
        <remap from="image_raw" to="/camera/image_raw"/>                
    </node>
              
</launch>
